ee/hogai/graph/memory/nodes.py

```diff
@@ -477,6 +495,74 @@ def _handle_remember_command(self, state: AssistantState) -> LangchainAIMessage
         else:
             return LangchainAIMessage(content="There's nothing to remember!", id=str(uuid4()))
 
+    def _handle_usage_command(self, state: AssistantState, config: RunnableConfig) -> AssistantMessage | None:
+        """
+        Handle the `/usage` slash command to show PostHog AI credit usage.
+        """
+        last_message = state.messages[-1] if state.messages else None
+        if not isinstance(last_message, HumanMessage) or last_message.content.strip() != SLASH_COMMAND_USAGE:
+            return None
+
+        # Check if the feature flag is enabled for this user
+        feature_flag_enabled = posthoganalytics.feature_enabled(
+            "posthog-ai-billing-usage-command",
+            str(self._user.distinct_id),
+            groups={"organization": str(self._team.organization_id)},
+            group_properties={"organization": {"id": str(self._team.organization_id)}},
+            send_feature_flag_events=False,
+        )
+
+        if not feature_flag_enabled:
+            # Feature is not enabled, return None to let it pass through as a normal query
+            return None
+
+        try:
+            # Get conversation ID from config
+            conversation_id = config.get("configurable", {}).get("thread_id")
+            if not conversation_id:
+                return AssistantMessage(content="Unable to retrieve conversation information.", id=str(uuid4()))
+
+            conversation_start = get_conversation_start_time(conversation_id)
+            if not conversation_start:
+                return AssistantMessage(content="Unable to retrieve conversation start time.", id=str(uuid4()))
+
+            past_month_start = get_past_month_start()
+            now = timezone.now()
+
+            # Calculate credits for current conversation
+            conversation_credits = get_ai_credits_for_team(
```

Comment:
Edge case: this doesn't take into account parallel conversations happening at the same time. Why can't we do a query by conversation, instead of using timestamps?